File: Blind75/230. Kth Smallest Element in a BST/230. Kth Smallest Element in a BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        vector<int> v;
        preorder(root, v);
        return v[k-1];
    }
    void preorder(TreeNode* node, vector<int>& v){
        if(node==NULL) return;
        preorder(node->left,v);
        v.push_back(node->val);
        preorder(node->right,v);
        
    }
};

File: Blind75/230. Kth Smallest Element in a BST/230. Kth Smallest Element in a BST.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    List<int> l = new List<int>();
    public int KthSmallest(TreeNode root, int k) {
        inorder(root);
        return l[k-1];
    }

    void inorder(TreeNode node){
        //base case
        if(node==null) return;
        inorder(node.left);
        l.Add(node.val);
        inorder(node.right);
    }
}
